query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java

```diff
@@ -243,37 +243,60 @@ private void handleConversionForArrayColumnExpression(
 
     if (handleConversionForNullOrEmptyArrayLiteral(lhs, operator, builder, value)) return;
 
-    // support only equals and IN operator
-    // both of them are handled as contains check to align with existing implementation
-    if (operator.equals("=")
-        || operator.equals("IN")
-        || operator.equals("!=")
-        || operator.equals("NOT IN")) {
-      // add NOT operator to negate the match condition
-      if (operator.equals("!=") || operator.equals("NOT IN")) {
+    // Support only equals and IN operator
+    if (operator.equals("=") || operator.equals("!=")) {
+      // Equals (=): CONTAINS(ip_types, 'Bot')
```

Comment:
Nit: changes looks fine. For readability, would it make sense to move to a function?